c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1107,19 +1148,35 @@ class DeviceCachingAllocator {
 
     AT_ASSERT(dst->is_split() && src->is_split());
 
-    if (dst->prev == src) {
+    if (dst->prev == src) { // [src dst]
       dst->ptr = src->ptr;
       dst->prev = src->prev;
       if (dst->prev) {
         dst->prev->next = dst;
       }
-    } else {
+
+      if (!dst->history) {
+        dst->history = std::move(src->history);
+        dst->history_last = src->history_last;
+      } else if (src->history) {
+        src->history_last->next = std::move(dst->history);
+        dst->history = std::move(src->history);
+      }
+      src->history_last = nullptr;
```

Comment:
Why reset only this one field?

Reply:
`src->history` is already moved/cleared above if it was set to something

Reply:
Ho right the move, ok!